File: fix_xcode_project.py
#!/usr/bin/env python3
"""
Script to add missing Swift files to the Xcode project
"""
import os
import re
import uuid
import sys

def generate_uuid():
    """Generate a 24-character hex UUID for Xcode"""
    return uuid.uuid4().hex[:24].upper()

def find_swift_files(root_dir):
    """Find all Swift files in the project"""
    swift_files = []
    for dirpath, dirnames, filenames in os.walk(root_dir):
        # Skip hidden directories and build directories
        dirnames[:] = [d for d in dirnames if not d.startswith('.') and d != 'Build']
        for filename in filenames:
            if filename.endswith('.swift'):
                full_path = os.path.join(dirpath, filename)
                relative_path = os.path.relpath(full_path, root_dir)
                swift_files.append((filename, relative_path))
    return swift_files

def is_file_in_project(pbxproj_content, filename):
    """Check if a file is already in the project"""
    return filename in pbxproj_content

def add_files_to_project(pbxproj_path, swift_files):
    """Add missing Swift files to the Xcode project"""
    
    with open(pbxproj_path, 'r') as f:
        content = f.read()
    
    # Find the main group ID and target ID
    main_group_match = re.search(r'mainGroup = ([A-F0-9]{24});', content)
    if not main_group_match:
        print("Could not find main group")
        return
    main_group_id = main_group_match.group(1)
    
    # Find the sources build phase
    sources_match = re.search(r'/\* Sources \*/ = \{\s+isa = PBXSourcesBuildPhase;[^}]+files = \([^)]+\);', content, re.DOTALL)
    if not sources_match:
        print("Could not find sources build phase")
        return
    
    # Track what we need to add
    files_to_add = []
    file_refs_to_add = []
    build_files_to_add = []
    
    for filename, relative_path in swift_files:
        if not is_file_in_project(content, filename):
            print(f"Adding {filename} to project...")
            
            # Generate UUIDs
            file_ref_id = generate_uuid()
            build_file_id = generate_uuid()
            
            # Create PBXFileReference
            file_ref = f"\t\t{file_ref_id} /* {filename} */ = {{isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = \"{filename}\"; sourceTree = \"<group>\"; }};"
            file_refs_to_add.append(file_ref)
            
            # Create PBXBuildFile
            build_file = f"\t\t{build_file_id} /* {filename} in Sources */ = {{isa = PBXBuildFile; fileRef = {file_ref_id} /* {filename} */; }};"
            build_files_to_add.append(build_file)
            
            # Track for adding to groups and build phases
            files_to_add.append((file_ref_id, build_file_id, filename))
    
    if not files_to_add:
        print("No files to add")
        return
    
    # Add PBXFileReferences
    file_ref_section = re.search(r'/\* Begin PBXFileReference section \*/(.*?)/\* End PBXFileReference section \*/', content, re.DOTALL)
    if file_ref_section:
        new_refs = '\n'.join(file_refs_to_add)
        content = content.replace(file_ref_section.group(0), 
                                  f"/* Begin PBXFileReference section */\n{file_ref_section.group(1)}{new_refs}\n/* End PBXFileReference section */")
    
    # Add PBXBuildFiles
    build_file_section = re.search(r'/\* Begin PBXBuildFile section \*/(.*?)/\* End PBXBuildFile section \*/', content, re.DOTALL)
    if build_file_section:
        new_builds = '\n'.join(build_files_to_add)
        content = content.replace(build_file_section.group(0),
                                  f"/* Begin PBXBuildFile section */\n{build_file_section.group(1)}{new_builds}\n/* End PBXBuildFile section */")
    
    # Add to main group children
    main_group_section = re.search(rf'{main_group_id}[^{{]*{{[^}}]*children = \(([^)]*)\);', content, re.DOTALL)
    if main_group_section:
        children = main_group_section.group(1)
        new_children = []
        for file_ref_id, _, filename in files_to_add:
            new_children.append(f"\t\t\t\t{file_ref_id} /* {filename} */,")
        
        new_children_str = '\n'.join(new_children)
        updated_children = f"{children}\n{new_children_str}"
        
        old_section = main_group_section.group(0)
        new_section = old_section.replace(children, updated_children)
        content = content.replace(old_section, new_section)
    
    # Add to sources build phase
    sources_section = re.search(r'(/\* Sources \*/ = \{[^}]+files = \()([^)]+)(\);)', content, re.DOTALL)
    if sources_section:
        files_list = sources_section.group(2)
        new_sources = []
        for _, build_file_id, filename in files_to_add:
            new_sources.append(f"\t\t\t\t{build_file_id} /* {filename} in Sources */,")
        
        new_sources_str = '\n'.join(new_sources)
        updated_files = f"{files_list}\n{new_sources_str}"
        
        content = content[:sources_section.start(2)] + updated_files + content[sources_section.end(2):]
    
    # Write back
    with open(pbxproj_path, 'w') as f:
        f.write(content)
    
    print(f"Added {len(files_to_add)} files to the project")

if __name__ == "__main__":
    project_dir = "/Users/nick/Documents/claude-code-ios-ui/ClaudeCodeUI-iOS"
    pbxproj_path = os.path.join(project_dir, "ClaudeCodeUI.xcodeproj", "project.pbxproj")
    
    # Find all Swift files
    swift_files = find_swift_files(project_dir)
    print(f"Found {len(swift_files)} Swift files")
    
    # Add missing files to project
    add_files_to_project(pbxproj_path, swift_files)
    print("Done!")

File: fix_xcode_project.rb
#!/usr/bin/env ruby

require 'xcodeproj'

# Path to the Xcode project
project_path = '/Users/nick/Documents/claude-code-ios-ui/ClaudeCodeUI-iOS/ClaudeCodeUI.xcodeproj'

puts "Opening project: #{project_path}"
project = Xcodeproj::Project.open(project_path)

# Get main target and UI test target
main_target = project.targets.find { |t| t.name == 'ClaudeCodeUI' }
ui_test_target = project.targets.find { |t| t.name == 'ClaudeCodeUIUITests' }

unless main_target
  puts "ERROR: Could not find main target 'ClaudeCodeUI'"
  exit 1
end

unless ui_test_target
  puts "ERROR: Could not find UI test target 'ClaudeCodeUIUITests'"
  exit 1
end

puts "Found main target: #{main_target.name}"
puts "Found UI test target: #{ui_test_target.name}"

# Track changes
removed_from_main = []
moved_to_uitest = []
removed_files = []

# Function to remove file from all targets
def remove_from_all_targets(file_ref, project)
  project.targets.each do |target|
    build_file = target.source_build_phase.files.find { |bf| bf.file_ref == file_ref }
    if build_file
      target.source_build_phase.remove_build_file(build_file)
      puts "  Removed from target: #{target.name}"
    end
  end
end

# Process all files in the project
puts "\nProcessing files..."
project.files.each do |file_ref|
  next unless file_ref.path
  
  file_name = File.basename(file_ref.path)
  
  # Check if file should be removed entirely (backup files, non-compilable files)
  if file_name.end_with?('.bak', '.bak11', '.bak12', '.bak13', '.bak14', '.bak15', '.bak16', '.bak17') ||
     file_name.end_with?('.md', '.txt', '.rb', '.sh', '.png', '.backup', '_backup.swift', '_FIXED.swift', '_Part2.swift', '_Extension.swift')
    
    puts "Removing backup/non-source file: #{file_ref.path}"
    remove_from_all_targets(file_ref, project)
    
    # Remove from project navigator
    file_ref.remove_from_project
    removed_files << file_name
    next
  end
  
  # Check if this is a UI test file that should be in UI test target
  if file_name.include?('UITest') || file_name.include?('UITests')
    # Check if it's in the main target
    main_build_file = main_target.source_build_phase.files.find { |bf| bf.file_ref == file_ref }
    
    if main_build_file
      puts "Found UI test file in main target: #{file_ref.path}"
      
      # Remove from main target
      main_target.source_build_phase.remove_build_file(main_build_file)
      removed_from_main << file_name
      
      # Add to UI test target if not already there
      ui_test_build_file = ui_test_target.source_build_phase.files.find { |bf| bf.file_ref == file_ref }
      unless ui_test_build_file
        ui_test_target.add_file_references([file_ref])
        moved_to_uitest << file_name
        puts "  Moved to UI test target"
      end
    end
  end
  
  # Special handling for specific problem files mentioned
  if ['ChatViewController_Part2.swift', 'ChatViewController_Extension.swift', 
      'ChatViewController_FIXED.swift', 'ChatViewController_backup.swift',
      'ViewControllers.swift'].include?(file_name)
    
    # These should not be in any target
    puts "Removing problematic file from all targets: #{file_ref.path}"
    remove_from_all_targets(file_ref, project)
    
    # Remove the file reference entirely if it's a backup or duplicate
    if file_name.include?('_backup') || file_name.include?('_FIXED') || 
       file_name.include?('_Part2') || file_name.include?('_Extension')
      file_ref.remove_from_project
      removed_files << file_name
    end
  end
end

# Look for specific UI test files that need to be moved
ui_test_files = ['AliveMainFlowUITests.swift', 'MajorFlowsUITests.swift']
ui_test_files.each do |filename|
  file_ref = project.files.find { |f| f.path && f.path.include?(filename) }
  
  if file_ref
    # Remove from main target if present
    main_build_file = main_target.source_build_phase.files.find { |bf| bf.file_ref == file_ref }
    if main_build_file
      puts "Removing #{filename} from main target"
      main_target.source_build_phase.remove_build_file(main_build_file)
      removed_from_main << filename
    end
    
    # Ensure it's in UI test target
    ui_test_build_file = ui_test_target.source_build_phase.files.find { |bf| bf.file_ref == file_ref }
    unless ui_test_build_file
      puts "Adding #{filename} to UI test target"
      ui_test_target.add_file_references([file_ref])
      moved_to_uitest << filename
    end
  end
end

# Clean up any nil or broken references
puts "\nCleaning up broken references..."
project.groups.each do |group|
  group.children.select { |child| child.nil? || (child.respond_to?(:path) && child.path.nil?) }.each do |broken|
    group.children.delete(broken) if broken
  end
end

# Remove duplicate references in build phases
[main_target, ui_test_target].each do |target|
  if target && target.source_build_phase
    seen_files = Set.new
    duplicates = []
    
    target.source_build_phase.files.each do |build_file|
      if build_file.file_ref && build_file.file_ref.path
        if seen_files.include?(build_file.file_ref.path)
          duplicates << build_file
        else
          seen_files.add(build_file.file_ref.path)
        end
      end
    end
    
    duplicates.each do |dup|
      target.source_build_phase.remove_build_file(dup)
      puts "Removed duplicate from #{target.name}: #{dup.file_ref.path if dup.file_ref}"
    end
  end
end

# Save the project
puts "\nSaving project..."
project.save

# Print summary
puts "\n" + "="*60
puts "SUMMARY OF CHANGES:"
puts "="*60

if removed_from_main.any?
  puts "\nRemoved from main target:"
  removed_from_main.each { |f| puts "  - #{f}" }
end

if moved_to_uitest.any?
  puts "\nMoved to UI test target:"
  moved_to_uitest.each { |f| puts "  - #{f}" }
end

if removed_files.any?
  puts "\nRemoved from project entirely:"
  removed_files.each { |f| puts "  - #{f}" }
end

puts "\nProject cleanup complete!"
puts "Next steps:"
puts "1. Open Xcode and build the project"
puts "2. If there are still issues, check the build log for specific errors"